### PR TITLE
Feature/shubham/search

### PR DESCRIPTION
**Pull request by panic-coder**

### CODE REVIEW

**Comment by kamal0808** [comment]
Just make the search case-insensitive and to be able to handle multiple identical skus

File: admin/src/app/home/orders/stock-orders/fulfill/fulfill.component.ts [hunk 1]
```diff
@@ -86,24 +86,29 @@ export class FulfillComponent implements OnInit {
...
 
 
-  getFulfilledStockOrderLineItems(limit?: number, skip?: number, productModelId?: string) {
+  getFulfilledStockOrderLineItems(limit?: number, skip?: number, productModelIds?: Array<string>) {
     if (!(limit && skip)) {
       limit = 100;
       skip = 0;
     }
-    if (!productModelId) {
+    if (!productModelIds) {
```
Comment (kamal0808) [inline admin/src/app/home/orders/stock-orders/fulfill/fulfill.component.ts]: also write condition for array length

File: admin/src/app/home/orders/stock-orders/fulfill/fulfill.component.ts [hunk 2]
```diff
@@ -147,22 +152,27 @@ export class FulfillComponent implements OnInit {
...
   }
 
-  getNotFulfilledStockOrderLineItems(limit?: number, skip?: number, productModelId?: string) {
+  getNotFulfilledStockOrderLineItems(limit?: number, skip?: number, productModelIds?: Array<string>) {
     if (!(limit && skip)) {
       limit = 100;
       skip = 0;
     }
-    if (!productModelId) {
+    if (!productModelIds) {
```
Comment (kamal0808) [inline admin/src/app/home/orders/stock-orders/fulfill/fulfill.component.ts]: array length check

File: admin/src/app/home/orders/stock-orders/generated/generated.component.ts [hunk 3]
```diff
@@ -101,18 +101,23 @@ export class GeneratedComponent implements OnInit, OnDestroy {
...
     }
     let sortOrder = this.sortAscending ? 'ASC' : 'DESC';
-    let filter = {
-      where: {
+    let whereFilter = {
         reportModelId: this.order.id,
-        approved: true,
-        productModelId: productModelId
-      },
+        approved: false
```
Comment (kamal0808) [inline admin/src/app/home/orders/stock-orders/generated/generated.component.ts]: why approved false?

File: admin/src/app/home/orders/stock-orders/receive/receive.component.ts [hunk 4]
```diff
@@ -89,25 +89,30 @@ export class ReceiveComponent implements OnInit, OnDestroy {
...
 
-  getReceivedStockOrderLineItems(limit?: number, skip?: number, productModelId?: string) {
+  getReceivedStockOrderLineItems(limit?: number, skip?: number, productModelIds?: Array<string>) {
 
     if (!(limit && skip)) {
       limit = 100;
       skip = 0;
     }
-    if (!productModelId) {
+    if (!productModelIds) {
```
Comment (kamal0808) [inline admin/src/app/home/orders/stock-orders/receive/receive.component.ts]: length check

File: admin/src/app/home/orders/stock-orders/receive/receive.component.ts [hunk 5]
```diff
@@ -150,22 +155,27 @@ export class ReceiveComponent implements OnInit, OnDestroy {
...
   }
 
-  getNotReceivedStockOrderLineItems(limit?: number, skip?: number, productModelId?: string) {
+  getNotReceivedStockOrderLineItems(limit?: number, skip?: number, productModelIds?: Array<string>) {
     if (!(limit && skip)) {
       limit = 100;
       skip = 0;
     }
-    if (!productModelId) {
+    if (!productModelIds) {
```
Comment (kamal0808) [inline admin/src/app/home/orders/stock-orders/receive/receive.component.ts]: length check

File: admin/src/app/home/orders/stock-orders/fulfill/fulfill.component.ts (hunk 1 above)
Comment (panic-coder) [reply admin/src/app/home/orders/stock-orders/fulfill/fulfill.component.ts]: done

File: admin/src/app/home/orders/stock-orders/fulfill/fulfill.component.ts (hunk 2 above)
Comment (panic-coder) [reply admin/src/app/home/orders/stock-orders/fulfill/fulfill.component.ts]: done

File: admin/src/app/home/orders/stock-orders/generated/generated.component.ts (hunk 3 above)
Comment (panic-coder) [reply admin/src/app/home/orders/stock-orders/generated/generated.component.ts]: done

File: admin/src/app/home/orders/stock-orders/receive/receive.component.ts (hunk 4 above)
Comment (panic-coder) [reply admin/src/app/home/orders/stock-orders/receive/receive.component.ts]: done

File: admin/src/app/home/orders/stock-orders/receive/receive.component.ts (hunk 5 above)
Comment (panic-coder) [reply admin/src/app/home/orders/stock-orders/receive/receive.component.ts]: done

File: admin/src/app/home/orders/stock-orders/complete/complete.component.ts [hunk 6]
```diff
@@ -45,17 +46,25 @@ export class CompleteComponent implements OnInit {
       });
   }
 
-  getStockOrderLineItems(limit?: number, skip?: number, productModelId?: string) {
+  getStockOrderLineItems(limit?: number, skip?: number, productModelIds?: Array<string>) {
     if (!(limit && skip)) {
       limit = 100;
       skip = 0;
     }
+    if ((productModelIds !== undefined && productModelIds !== null) && (!productModelIds && productModelIds.length > 0)) {
```
Comment (kamal0808) [inline admin/src/app/home/orders/stock-orders/complete/complete.component.ts]: fix this condition

File: nginx.warehouse.conf [hunk 7]
```diff
@@ -13,7 +13,7 @@ upstream notification-service {
 }
 server {
     listen 80 default;
-    server_name _;
+    server_name shrey.shoppinpal.com;
```
Comment (kamal0808) [inline nginx.warehouse.conf]: remove

**Pull request merged**